### PR TITLE
[4.3][Guided Tours] Administrator console unresponsive when CSS selector is invalid

### PR DESCRIPTION
**Pull request by obuisard**
When someone creating a tour enters a selector that is not valid, the tour does not start, produces an error in the browser console, nothing happens but the tour keeps trying to start until the user logs out. The user receives no feedback, unless he/her is aware of the browser console.

### Summary of Changes

We already caught the error but the session storage was not cleared.
This PR clears the storage on error and produces an error message for the user.

### Testing Instructions

Create a step or edit a step in a tour. Enter an invalid target, for instance `#1_tada`. This selector is invalid because it starts with a number.
Start the tour. you should see the error message:

image

### Actual result BEFORE applying this Pull Request

The tour does not start, there is no user feedback and the user needs to logout to recover.

### Expected result AFTER applying this Pull Request

The tour does not start, the administrator console is unaffected, a message pops up explaining that the tour cannot run and the user can resume activities.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 99b66de2c7e9009b29b34050cd6d405a2c8efc6c<br /><br />not a fan of the tone of the message but can confirm that it does what it says<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40054">issues.joomla.org/tracker/joomla-cms/40054</a>.</sub>

File: build/media_source/plg_system_guidedtours/js/guidedtours.es6.js [hunk 1]
```diff
@@ -387,7 +387,12 @@ function loadTour(tourId) {
         startTour(result.data);
       })
       .catch((error) => {
-        throw new Error(error);
+        // Kill the tour if there is a problem with selector validation
+        emptyStorage();
+        console.log(`Exception error - ${error.message} - Tour stopped`);
```
Comment (laoneo) [inline build/media_source/plg_system_guidedtours/js/guidedtours.es6.js]: Thanks you for the pull request. Unfortunately I don't think so we should log to the console of the browser. Do we do than nowadays?

Comment (Fedik) [reply build/media_source/plg_system_guidedtours/js/guidedtours.es6.js]: This is okay, to help to find a real reason of error.
But please log it without decoration. Make it:
```javascript
console.error(error);
```

Comment (dgrammatiko) [reply build/media_source/plg_system_guidedtours/js/guidedtours.es6.js]: What Allon said, use `throw new Error('Error message')`

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: build/media_source/plg_system_guidedtours/js/guidedtours.es6.js (hunk 1 above)
Comment (obuisard) [reply build/media_source/plg_system_guidedtours/js/guidedtours.es6.js]: Thanks

File: build/media_source/plg_system_guidedtours/js/guidedtours.es6.js [hunk 2]
```diff
@@ -387,9 +387,10 @@ function loadTour(tourId) {
         startTour(result.data);
       })
       .catch((error) => {
+        throw new Error(error);
```
Comment (dgrammatiko) [inline build/media_source/plg_system_guidedtours/js/guidedtours.es6.js]: After the throw nothing else is executed, thus this line needs to move to 396 after `Joomla.renderMessages(messages);`

Comment (obuisard) [reply build/media_source/plg_system_guidedtours/js/guidedtours.es6.js]: You're right, I was not thinking...

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on c197ca8c0aff65680ffa3bb3ba3c4831bc6ec7e7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40054">issues.joomla.org/tracker/joomla-cms/40054</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on c197ca8c0aff65680ffa3bb3ba3c4831bc6ec7e7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40054">issues.joomla.org/tracker/joomla-cms/40054</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40054">issues.joomla.org/tracker/joomla-cms/40054</a>.</sub>

**Pull request merged**